tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,296 @@
+        );
+    }
+
+    /// @notice Asserts that the SuperchainConfig is setup correctly
+    function checkSuperchainConfig() internal view {
+        console.log("Running chain assertions on the SuperchainConfig");
+
+        require(proxies.SuperchainConfig.code.length != 0, "7100");
+
+        SuperchainConfig superchainConfig = SuperchainConfig(proxies.SuperchainConfig);
+
+        require(superchainConfig.guardian() == cfg.superchainConfigGuardian(), "7200");
+        require(superchainConfig.paused() == false, "7300");
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck() internal view override {
+        console.log("Running post-deploy assertions");
+        ResourceMetering.ResourceConfig memory actualResourceConfig =
+            SystemConfig(proxies.SystemConfig).resourceConfig();
+        ResourceMetering.ResourceConfig memory expectedResourceConfig = Constants.DEFAULT_RESOURCE_CONFIG();
+        require(keccak256(abi.encode(actualResourceConfig)) == keccak256(abi.encode(expectedResourceConfig)), "100");
+
+        checkSystemConfig();
+        checkL1CrossDomainMessenger();
+        checkL1StandardBridge();
+        checkL2OutputOracle();
+        checkOptimismMintableERC20Factory();
+        checkL1ERC721Bridge();
+        checkOptimismPortal();
+        checkProtocolVersions();
+        checkSuperchainConfig();
+    }
+
+    /// @notice Reads the contract addresses from lib/superchain-registry/superchain/extra/addresses/mainnet/op.json
+    function _getContractSet() internal returns (Types.ContractSet memory _proxies) {
+        string memory json;
+
+        // Read op.json
+        try vm.readFile(
+            string.concat(vm.projectRoot(), "/lib/superchain-registry/superchain/extra/addresses/mainnet/op.json")
+        ) returns (string memory data) {
+            json = data;
+        } catch {
+            revert("Failed to read lib/superchain-registry/superchain/extra/addresses/mainnet/op.json");
+        }
+
+        _proxies.L1CrossDomainMessenger = stdJson.readAddress(json, "$.L1CrossDomainMessengerProxy");
+        _proxies.L1StandardBridge = stdJson.readAddress(json, "$.L1StandardBridgeProxy");
+        _proxies.L2OutputOracle = stdJson.readAddress(json, "$.L2OutputOracleProxy");
+        _proxies.OptimismMintableERC20Factory = stdJson.readAddress(json, "$.OptimismMintableERC20FactoryProxy");
+        _proxies.OptimismPortal = stdJson.readAddress(json, "$.OptimismPortalProxy");
+        _proxies.OptimismPortal2 = stdJson.readAddress(json, "$.OptimismPortalProxy");
+        _proxies.SystemConfig = stdJson.readAddress(json, "$.SystemConfigProxy");
+        _proxies.L1ERC721Bridge = stdJson.readAddress(json, "$.L1ERC721BridgeProxy");
+
+        // Read superchain.yaml
+        string[] memory cmds = new string[](6);
+
+        // Build ffi command string
+        cmds[0] = "cat";
+        cmds[1] = "superchain.yaml";
+        cmds[2] = "|";
+        cmds[3] = "yq";
```

Comment:
We will need to add `yq` to the required deps in the `README` if we agree on using it